### PR TITLE
feat: ci: packer snap

### PR DESCRIPTION
**Pull request by coryschwartz**
## Checklist

Before you mark the PR ready for review, please make sure that:
- [ ] All commits have a clear commit message.
- [ ] The PR title is in the form of of `<PR type>: <#issue number> <area>: <change being made>`
    - example: ` fix: #1234 mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_, _misc_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

**Comment by coryschwartz** [comment]
This changes the snap package so it can be used on digitalocean as a service

The snap package will install a service rather than just commandline tools, meaning lotus daemon is running in the background, enabled at boot.

commandline tools for the miner, worker, and lotus are of course installed as part of the snap. The lotus commandline is configured to interact with the local daemon by default, and can be made to use another daemon or gateway with environment variables, as usual.

Included in this PR is a change to the digitalocean image, this will make a digitalocean image install the snap package rather than copying the lotus binaries. This way, we do not need to update the digitalocean image when there are new releases. Simply by updating the snap package, the digitalocean images will start using the updated snap.

**Comment by coryschwartz** [comment]
If you'd like to try out the images created with this change, see lotus-mainnet-snap-20220119015458 in the DO account.

File: .circleci/config.yml [hunk 1]
```diff
@@ -705,6 +705,17 @@ jobs:
       - packer/build:
           template: tools/packer/lotus.pkr.hcl
           args: "-var ci_workspace_bins=./linux-butterflynet -var lotus_network=butterflynet -var git_tag=$CIRCLE_TAG"
+  publish-packer-snap:
```
Comment (magik6k) [inline .circleci/config.yml:708]: Changes in this file need to match `template.yml`

**Comment by coryschwartz** [comment]
Sorry for the delay, should be good to go now.

**Review by magik6k — APPROVED** [review]

**Pull request merged**